demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -766,19 +810,43 @@ static int32_t sha256Final( void * hashContext,
                             char * pOutput,
                             size_t outputLen )
 {
+    assert( outputLen >= SHA256_HASH_DIGEST_LENGTH );
+
     ( void ) outputLen;
+
     return mbedtls_sha256_finish_ret( ( mbedtls_sha256_context * ) hashContext,
                                       ( unsigned char * ) pOutput );
 }
 
+static void getHeaderStartLocFromHttpRequest( HTTPRequestHeaders_t requestHeaders,
+                                              char ** pStartHeaderLoc,
+                                              size_t * startHeaderLocLen )
+{
+    size_t headerLen = requestHeaders.headersLen;
+    char * pHeaders = ( char * ) requestHeaders.pBuffer;
+
+    assert( pStartHeaderLoc != NULL );
+    assert( startHeaderLocLen != NULL );
+
+    while( pHeaders != NULL && headerLen >= 2 && 0 != strncmp( pHeaders, "\r\n", strlen( "\r\n" ) ) )
+    {
+        pHeaders++;
+        headerLen--;
+    }
+
```

Comment:
There should be an assert failure if the `\r\n` string isn't found in the request data.